pkg/sqlx/querybuilder_performer.go

```diff
@@ -532,25 +532,29 @@ func (qb *performerQueryBuilder) GetAllPiercings(ids []uuid.UUID) ([][]*models.B
 
 func (qb *performerQueryBuilder) SearchPerformers(term string, limit int) (models.Performers, error) {
 	query := `
-	SELECT P.* FROM performers P
-	JOIN (
-		SELECT DISTINCT ON (P.id)
-			P.id,
-			(similarity(P.name, $1) +
-			 similarity(COALESCE(PA.alias, ''), $1) * 0.7 +
-			 similarity(COALESCE(P.disambiguation, ''), $1) * 0.3
-			) AS score
-		FROM performers P
-		LEFT JOIN performer_aliases PA on PA.performer_id = P.id
-		WHERE
-			P.deleted = FALSE AND (
-				(P.name % $1 AND similarity(P.name, $1) > 0.5)
-				OR (PA.alias % $1 AND similarity(COALESCE(PA.alias, ''), $1) > 0.6)
-				OR (P.disambiguation % $1 AND similarity(COALESCE(P.disambiguation, ''), $1) > 0.7)
-			)
-	) T ON T.id = P.id
-	ORDER BY T.score DESC
-	LIMIT $2`
+		SELECT P.* FROM performers P
+		WHERE P.id IN (
+			SELECT id FROM (
+				SELECT P.id, similarity(P.name, $1) AS similarity
+				FROM performers P
+				LEFT JOIN performer_aliases PA on PA.performer_id = P.id
+				WHERE P.deleted = FALSE AND (P.name % $1 AND similarity(P.name, $1) > 0.5)
+			UNION
+				SELECT P.id, (similarity(COALESCE(PA.alias, ''), $1) * 0.7) AS similarity
+				FROM performers P
+				LEFT JOIN performer_aliases PA on PA.performer_id = P.id
+				WHERE P.deleted = FALSE AND (PA.alias % $1 AND similarity(COALESCE(PA.alias, ''), $1) > 0.6)
+			UNION
+				SELECT P.id, (similarity(COALESCE(P.disambiguation, ''), $1) * 0.3) AS similarity
+				FROM performers P
+				LEFT JOIN performer_aliases PA on PA.performer_id = P.id
+				WHERE P.deleted = FALSE AND (P.disambiguation % $1 AND similarity(COALESCE(P.disambiguation), $1) > 0.7)
+			) A
+			GROUP BY id
+			ORDER BY SUM(similarity)
```

Comment:
I've hit a similar issue my first try:
You need the order to be on the final query, otherwise it doesn't keep that order.
That's why the test is failing too - it searches for some `performer-{int}`, and that should be the first match because it is an exact match.

Reply:
well, correction, I'm not 100% sure it's the same issue because the query is totally different from what I tried, but it's a possibility.